Makefile

```diff
@@ -26,7 +26,13 @@ OS     ?= $(shell go env GOOS)
 
 HELM_VERSION ?= 3.12.3
 KUBEBUILDER_TOOLS_VERISON ?= 1.28.0
+KUBECTL_VERSION ?= 1.28.2
+KIND_VERSION ?= $(shell grep "sigs.k8s.io/kind" go.mod | awk '{print $$NF}')
 GINKGO_VERSION ?= $(shell grep "github.com/onsi/ginkgo/v2" go.mod | awk '{print $$NF}')
+HELM_DOCS_VERSION ?= $(shell grep "github.com/norwoodj/helm-docs" hack/tools/go.mod | awk '{print $$NF}')
+BOILERSUITE_VERSION ?= $(shell grep "github.com/cert-manager/boilersuite" hack/tools/go.mod | awk '{print $$NF}')
+CONTROLLER_TOOLS_VERSION ?= $(shell grep "sigs.k8s.io/controller-tools" hack/tools/go.mod | awk '{print $$NF}')
```

Comment:
Is the go.mod in hack/tools really useful anymore? Could it make sense just to have the versions mastered from the Makefile, or is it used for something else?

I suspect this problem might be present in other cert-manager projects...

Reply:
I really strongly dislike hack/tools/go.mod for so many reasons:

- Introduces the risk of dependency conflicts between tools we depend on
- Builds tools with dependency versions from `hack/tools/go.mod` which might not be versions they were actually tested with
- Creates a confusing phantom module in the repo
- Noiser to update - updates can change dependencies / `go.sum`, as opposed to a single line version bump
- We still have to have another way of managing tools, because not all tools we use are go based - so it ends up being a second mechanism for tracking tools, not a unifying force

The upside is that `go get -u` makes it slightly easier to bump go-based tools to the latest version, but that upside isn't close to being worth it to me. `go install` is much better for this sort of thing.

I'd love to get rid of hack/tools everywhere, but because it's [in the golang/go wiki](https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module) I've not been able to successfully argue for its removal. (Obviously I disagree with the wiki on this)

If we could make the consensus be that we don't want to use it anywhere I'd gladly sign up for that and I think it would be an improvement across the board. But I also don't want to have to fight that fight to get this PR merged!